src/main/java/org/broadinstitute/hellbender/tools/walkers/qc/CheckPileup.java

```diff
@@ -0,0 +1,226 @@
+        ReadPileup pileup = context.getBasePileup();
+        SAMPileupFeature truePileup = getTruePileup( tracker );
+
+        if ( truePileup == null ) {
+            out.printf("No truth pileup data available at %s%n", pileup.getPileupString((char) ref.getBase()));
+            if ( !continueAfterAnError) {
+                throw new UserException.BadInput(String.format("No pileup data available at %s given GATK's output of %s -- this walker requires samtools pileup data over all bases",
+                        context.getLocation(), new String(pileup.getBases())));
+            }
+        } else {
+            String pileupDiff = pileupDiff(pileup, truePileup, true);
+            if ( pileupDiff != null ) {
+                out.printf("%s vs. %s%n", pileup.getPileupString((char) ref.getBase()), truePileup.getPileupString());
+                if ( !continueAfterAnError) {
+                    throw new UserException.BadInput(String.format("The input pileup doesn't match the GATK's internal pileup: %s", pileupDiff));
+                }
+            }
+        }
+
+        nLoci++;
+        nBases += pileup.size();
+    }
+
+    private static String maybeSorted( final String x, boolean sortMe )
+    {
+        if ( sortMe ) {
+            byte[] bytes = x.getBytes();
+            Arrays.sort(bytes);
+            return new String(bytes);
+        }
+        return x;
+    }
+
+    public String pileupDiff(final ReadPileup a, final SAMPileupFeature b, boolean orderDependent)
+    {
+        if ( a.size() != b.size() ) {
+            return "Sizes not equal";
+        }
+        if (IntervalUtils.compareLocatables(a.getLocation(), b, getReferenceDictionary()) != 0 ) {
+            return "Locations not equal";
+        }
+
+        String aBases = maybeSorted(new String(a.getBases()), ! orderDependent );
+        String bBases = maybeSorted(b.getBasesAsString(), ! orderDependent );
+        if ( ! aBases.toUpperCase().equals(bBases.toUpperCase()) ) {
+            return "Bases not equal";
+        }
+
+        String aQuals = maybeSorted(new String(a.getBaseQuals()), ! orderDependent );
+        String bQuals = maybeSorted(new String(b.getQuals()), ! orderDependent );
+        if ( ! aQuals.equals(bQuals) ) {
+            return "Quals not equal";
+        }
+
+        return null;
+    }
+
+    /**
+     * Extracts the true pileup data from the given rodSAMPileup.  Note that this implementation
+     * assumes that the genotype will only be point or indel.
+     * @param tracker ROD tracker from which to extract pileup data.
+     * @return True pileup data.
+     */
+    private SAMPileupFeature getTruePileup( FeatureContext tracker ) {
+        final List<SAMPileupFeature> features = tracker.getValues(pileup);
+        SAMPileupFeature pileupArg = (features.isEmpty()) ? null : features.get(0);
+
+        if( pileupArg == null)
+            return null;
+
+        if( pileupArg.hasPointGenotype() ) {
```

Comment:
this logic is strange, getPointGenotype just returns  `isPointGenotype() ? this : null` and so does getIndelGenotypes

and it looks like `hasPointGenotype` is defined as `!hasIndelGenotype`

I think this whole block can just be `return pileupArg == null ? null : return pileupArg`  

I don't see how that exception could ever be thrown...  Maybe I'm just missing something dumb though.